packages/storybook-builder-vite/index.ts

```diff
@@ -16,25 +15,16 @@ export interface ViteStats {}
 
 export type ViteBuilder = Builder<UserConfig, ViteStats>;
 
-function parseRequest(id: string): Record<string, string> | null {
-  const search = id.split('?').pop();
-  if (!search) {
-    return null;
-  }
-
-  return Object.fromEntries(new URLSearchParams(search));
-}
-
 function iframeMiddleware(options: ExtendedOptions, server: ViteDevServer): RequestHandler {
   return async (req, res, next) => {
-    if (!req.url.match(/^\/iframe.html($|\?)/)) {
+    if (!req.url.match(/^\/iframe\.html($|\?)/)) {
       next();
       return;
     }
 
-    const query = parseRequest(req.url)!;
-
-    if (query['html-proxy'] !== undefined) {
+    // We need to handle `html-proxy` params for style tag HMR https://github.com/eirslett/storybook-builder-vite/issues/266#issuecomment-1055677865
+    // e.g. /iframe.html?html-proxy&index=0.css
+    if (req.query['html-proxy'] !== undefined) {
```

Comment:
I kept this separate to the check above for readability. Happy to merge them and move the comment if it's preferred.

Reply:
It's good this way, I think.